src/segyio/segy.c

```diff
@@ -415,6 +415,10 @@ int segy_set_bfield( char* binheader, int field, int val ) {
 }
 
 int segy_binheader( FILE* fp, char* buf ) {
+    if(fp == NULL) {
```

Comment:
Maybe we should introduce a new error code (SEGY_INVALID_ARGS or something) for when a function cannot complete because its arguments are garbage? This would lead to errno getting checked, and errno will be success.